### PR TITLE
Reverse Lookup in 1D array in cpp

### PR DESCRIPTION
**Pull request by aromal17**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I have added the code for reverse lookup in 1D array and generated the pull request

### CODE REVIEW

**Comment by aromal17** [comment]
I have added the code for reverse lookup in 1D and updated the readme file for the same
Please review it and kindly let me know of any changes so that I can proceed with another pull request for reverse lookup in 2D array.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 
- Modularize the beginner description. Its too large to consume easily
- Add description of functions 
- Add a new line at the EOF
- Use a Code formatter to format the Code

**Comment by aromal17** [comment]
I have made the requested changes and committed the same...Kindly review the changes and please let me know of any further changes.

File: C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Input : 
+6
+2 3 4 7 8 9
+
+Output:
+308
+
+Time Complexity : O(n)
+Space Complexity : O(1)
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp]: Add a new line at the EOF

File: C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+int main()
+{
+    int n, value;
+    long long total_sum;
+    vector<int> array;
+    cin >> n; 
+  
+    // reading the elements to the array
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp]: Redundant Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aromal17** [comment]
I have removed the redundant comment and added a new line at the EOF. Kindly review the code for possible improvisations.

File: C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Output:
+308
+
+Time Complexity : O(n)
+Space Complexity : O(1)
+*/
+
+
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp]: Remove extra lines of code

File: C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+-> (a * b) % M = ((a % M) * (b % M)) % M
+where, M = 10^9 + 7
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+const long long M = 1e9 + 7;
+
+// function subset_sums() calculates the total sum of all subsets. 
+// contribution variable denotes the contribution given by each element in the array.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Reverse_Lookup_in_1D_array.cpp]: Use Multi-Line Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aromal17** [comment]
I have made the requested changes to the file. Kindly review the code.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**